### PR TITLE
Merge 2.6.x -> 3.2.x

### PR DESCRIPTION
**Pull request by mwaggett**

### CODE REVIEW

**Review by Magisus — APPROVED** [review]

**Comment by mwaggett** [comment]
don't merge yet

**Comment by mwaggett** [comment]
adding pupperware as CODEOWNERS

**Comment by mwaggett** [comment]
okay ready now

**Pull request merged**
